### PR TITLE
Setup the official documentation

### PR DESCRIPTION
**Pull request by mrproliu**

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please move the image to the website to host.

**Comment by mrproliu** [comment]
Sure, should I also add the documentation link to the website?

**Comment by wu-sheng** [comment]
The link is only working after merging.

File: docs/en/setup/configuration/process_discovery/vm.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# VM Process Detector
+
+The VM process detector could be detected any process in the VM with finders.
```
Comment (wu-sheng) [inline docs/en/setup/configuration/process_discovery/vm.md:3]:
```suggestion
The VM process detector could detect any process in the VM with finders.
```

File: docs/en/setup/configuration/process_discovery/vm.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Configuration
+
+| Name | Default | Environment Key | Description |
+|------|---------|-----------------|-------------|
+| process_discovery.vm.active | false | ROVER_PROCESS_DISCOVERY_VM_ACTIVE | Is active the VM mode to detect processes. |
+| process_discovery.vm.period | 3s | ROVER_PROCESS_DISCOVERY_VM_PERIOD | The period to detect the process. |
+| process_discovery.vm.finders | | | It could be define multiple finders to find the process. It only provide one finder by default. |
+| process_discovery.vm.finders.match_cmd_regex | | ROVER_PROCESS_DISCOVERY_VM_FINDER_MATCH_CMD_REGEX | Use regex string to locate the process from the command line of the process. |
+| process_discovery.vm.finders.layer | OS_LINUX | ROVER_PROCESS_DISCOVERY_VM_FINDER_LAYER | The Layer need to relate to the process entity |
```
Comment (wu-sheng) [inline docs/en/setup/configuration/process_discovery/vm.md:14]: Replace all `need to` to `define`/`declare` or something.

File: docs/en/setup/configuration/process_discovery/vm.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+The metadata of the process could build by the Go Template to help dynamically build them, also, you could just configure it as the string value, and it still works.
+
+These fields are supported using template to build:
+1. Service Name
+2. Service Instance Name
+3. Process NAme
+
+### Context
+
+The context provided multiple functions for helping you build the process metadata.
```
Comment (wu-sheng) [inline docs/en/setup/configuration/process_discovery/vm.md:36]: `provided` -> `provides`, don't use pass-tense。

File: docs/en/setup/examples/deploy/linux/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Deploy on Linux
+
+It could help you run the Rover in Linux.
```
Comment (wu-sheng) [inline docs/en/setup/examples/deploy/linux/readme.md:3]:
```suggestion
```

File: docs/en/setup/examples/cases/profiling-with-vm-process/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+1. Active the VM Process Detector.
+2. Update the finder configuration.
+
+You could be following [this configuration](../../../configuration/process_discovery/vm.md) to get more configuration information.
+
+### Enable Profiling
+
+Make sure the profiling module has been active.
+
+You could be following [this configuration](../../../configuration/profiling.md) to get more configuration information.
```
Comment (wu-sheng) [inline docs/en/setup/examples/cases/profiling-with-vm-process/readme.md:19]: Please add a real example, such as providing a golang/C service running, and how to identify it.

File: docs/en/setup/overview.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+You can quickly build your Rover according to the following examples:
+
+### Deploy
+
+1. [Deploy on Linux](examples/deploy/linux/readme.md)
+
+### Use Cases
+
+1. [Profiling with VM Process](examples/cases/profiling-with-vm-process/readme.md)
```
Comment (wu-sheng) [inline docs/en/setup/overview.md:28]:
```suggestion
1. [Profiling a golang/C service on Linux](examples/cases/profiling-with-vm-process/readme.md)
```

File: docs/menu.yml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+          path: /en/setup/overview
+        - name: Configuration
+          catalog:
+            - name: Core Module
+              path: /en/setup/configuration/core
+            - name: Process Discovery Module
+              catalog:
+                - name: Overview
+                  path: /en/setup/configuration/process_discovery/overview
+                - name: VM Process Detector
```
Comment (wu-sheng) [inline docs/menu.yml:44]:
```suggestion
                - name: Linux Process Detector
```

File: docs/en/concepts-and-designs/module_design.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# Module Design
+## Overview
+
+The module is an isolation concept in Rover. Each module completes an independent future.
```
Comment (kezhenxu94) [inline docs/en/concepts-and-designs/module_design.md:4]:
```suggestion
The module is an isolation concept in Rover. Each module completes an independent feature.
```

File: docs/en/concepts-and-designs/module_design.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+## Life Cycle
+
+Each concept has a complete life cycle.
+
+- Start: start phase is to start the current module.
```
Comment (kezhenxu94) [inline docs/en/concepts-and-designs/module_design.md:10]:
```suggestion
- Start: Start phase is to start the current module.
```

File: docs/en/concepts-and-designs/module_design.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+- NotifyStartSuccess: Execute when all modules have finished starting without any errors.
+- Shutdown: The shutdown phase is to close all the used resources.
+
+## Config
+
+Each module corresponds to configuration data and only set in the configuration file, it would be enabled.
```
Comment (kezhenxu94) [inline docs/en/concepts-and-designs/module_design.md:16]:
```suggestion
Each module has its corresponding configurations and only when they're set in the configuration file, the module would be enabled.
```

File: docs/en/concepts-and-designs/overview.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+SkyWalking Rover is an open-source collector, which provides a metrics collector and eBPF-based profiler for C, C++, and Golang.
+
+## Why use SkyWalking Rover?
+
+On the Linux platform, we could collect a lot of telemetry data. Rover could collect them based on the eBPF technology,
+and upload them to the SkyWalking backend for analysis, aggregate, and visualize them.
+
+1. EBPF based profiling for C, C++, and Golang.
+2. Infrastructure metrics collector for VM, Process, etc.
+3. Optimized the network performance.
```
Comment (kezhenxu94) [inline docs/en/concepts-and-designs/overview.md:12]: If this is not supported (yet) I'd just remove it

File: docs/en/guide/compile/how-to-compile.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+# Compiling
+
+## Go version
+
+Go version `1.17` or higher is supported for compilation.
+
+## Platform
+Linux is supported in the SkyWalking Rover.
```
Comment (kezhenxu94) [inline docs/en/guide/compile/how-to-compile.md:8]: Linux is not supported out of box, you need to specify

- Minimum version of Linux kernel
- What dependencies they need to install `apt install ...`

Comment (mrproliu) [reply docs/en/guide/compile/how-to-compile.md]: Updated, please recheck.

File: docs/en/setup/examples/cases/profiling-with-vm-process/readme.md (hunk 5 above)
Comment (mrproliu) [reply docs/en/setup/examples/cases/profiling-with-vm-process/readme.md]: Added, please recheck.

File: docs/en/setup/configuration/process_discovery/vm.md (hunk 2 above)
Comment (mrproliu) [reply docs/en/setup/configuration/process_discovery/vm.md]: Updated.

File: docs/en/setup/examples/cases/profiling-process/readme.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
+# Profiling a golang/C service on Linux
```
Comment (wu-sheng) [inline docs/en/setup/examples/cases/profiling-process/readme.md:1]:
```suggestion
# Profiling a Golang/C service on Linux
```

File: docs/en/setup/examples/cases/profiling-process/readme.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+        instance_name: {{.Rover.HostIPV4 "en0"}}
+        process_name: {{.Process.ExeName}}
+        labels: ""
+```
+
+You could be following [this configuration](../../../configuration/process_discovery/linux.md) to get more configuration information.
+
+### Enable Profiling
+
+Make sure the profiling module has been active.
```
Comment (wu-sheng) [inline docs/en/setup/examples/cases/profiling-process/readme.md:65]: Whether the above is full configuration? Why with a `enable` condition?

I think you should post a real complete configuration here.

Comment (mrproliu) [reply docs/en/setup/examples/cases/profiling-process/readme.md:65]: Already post the complete configuration below. The process discovery and profiling are two separate modules, so I used two chapters and configuration to describe them.

Comment (mrproliu) [reply docs/en/setup/examples/cases/profiling-process/readme.md]: Updated.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**